### ISSUE TITLE
iOS: can not `pause()` after `setSpeed()`

### ISSUE THREAD

**Issue by yhcharles** [ryanheise/just_audio#163]
**Which API doesn't behave as documented, and how does it misbehave?**
setSpeed()

**Minimal reproduction project**
Provide a link here using one of two options:
1. Fork this repository and modify the example to reproduce the bug, then provide a link here.
2. If the unmodified official example already reproduces the bug, just write "The example".

**To Reproduce (i.e. user steps, not code)**

iOS:
1. create a player: `AudioPlayer()`
2. set an audio file: `setFilePath()`
3. set speed: `setSpeed()`, now it will start playing even without calling `play()`, is this expected?
4. try to call `pause()`, but it doesn't work, the audio is still playing

**Error messages**

```
If applicable, copy & paste error message here, within the triple quotes to preserve formatting.
```

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. MacOS + version]
 - Browser [e.g. chrome, safari + version]

**Smartphone (please complete the following information):**
 - Device: iPhone Xs
 - OS: iOS 13.5.1

**Flutter SDK version**
```
insert output of "flutter doctor" here
```

**Additional context**

**Comment by ryanheise** [comment]
> **Minimal reproduction project**
> Provide a link here using one of two options:
> 
>     1. Fork this repository and modify the example to reproduce the bug, then provide a link here.
> 
>     2. If the unmodified official example already reproduces the bug, just write "The example".

This should be reproducible on the example.

**Comment by ryanheise** [comment]
I have just fixed it on git master. Please let me know how it goes for you.

**Comment by yhcharles** [comment]
Yes, it works now. Thank you for the quick response and fix!

**Comment by ryanheise** [comment]
Good to hear! Unfortunately I can't publish the fix just yet, due to an issue with pub.dev (https://github.com/flutter/flutter/issues/52285 and https://github.com/dart-lang/pub-dev/issues/3971), but you can point your pubspec to the git hash for this commit and use it that way.

**Comment by ryanheise** [comment]
Just to update you, the issue with pub.dev has been fixed and I've been able to publish release 0.4.1 .

**Comment by yhcharles** [comment]
> Just to update you, the issue with pub.dev has been fixed and I've been able to publish release 0.4.1 .

Awesome, thank you!
